src/platform/SDL2Window.cxx

```diff
@@ -187,128 +181,110 @@ void SDLWindow::swapBuffers()
 bool SDLWindow::create(void)
 {
     int targetWidth, targetHeight;
-    getSize(targetWidth, targetHeight);
-    SDL_bool windowWasGrabbed = SDL_FALSE;
-    if (windowId != NULL)
-        windowWasGrabbed = SDL_GetWindowGrab(windowId);
-
-    // if we have an existing identical window, go no further
-    if (windowId != NULL)
+    // If fullscreen, the target size will be that of the selected fullscreen resolution
+    if (fullScreen)
+        const_cast<SDLDisplay *>((const SDLDisplay *)getDisplay())->getWindowSize(targetWidth, targetHeight);
```

Comment:
```suggestion
        const_cast<SDLDisplay *>(static_cast<const SDLDisplay *>(getDisplay()))->getWindowSize(targetWidth, targetHeight);
```
it's odd to mix C-style (bad) and C++-style (less bad) casts in a single statement.

And `getWindowSize` isn't a const method? Why not?